### ISSUE TITLE
lint to protect against SQL injection (package:sqflite)

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#2086]
**Ensure that SQL query strings are always constant, or use whitelisted string-interpolation methods.**

Specifically guard against unsafe use of [package:sqflite](https://github.com/tekartik/sqflite/tree/master/sqflite).

_(Placeholder.  Details to follow.)_

**Comment by mehmetf** [comment]
I am a bit skeptical about this living in the linter itself.

We want library specific lints but I believe these should live in the library itself. The library owner would want some level of control over these lints.

**Comment by bwilkerson** [comment]
I agree that such a lint would be best bundled with the package with which it is associated.

Unfortunately, there's more friction involved when choosing that path. Enough so that we usually end up not implementing such lints. (I'm happy to have discuss the reasons with you if you're interested.)

However, being extremely pragmatic, we could define the rule in the linter and then work with the package author move it to the package at some future point.
